src/containers/topbutton/Top.css

```diff
@@ -0,0 +1,21 @@
+#topButton {
+  visibility:hidden;
+  position: fixed;
+  bottom: 20px;
+  right: 25px;
+  opacity: 0.5;
+  z-index: 99;
+  border: none;
+  outline: none;
+  color: #0a4c61;
+  cursor: pointer;
+  padding: 20px;
+  border-radius: 20px;
+  font-size: 15px;
+}
+
+#topButton:hover {
+  background-color: #081e25;
```

Comment:
Pass the `theme` as prop to this component. and use the colors from the theme. i.e. `theme.text` to body of button, `theme.body` to arrow color and `theme.text` should be border of the button. Also, note that onHover, border color must remain same (theme.text), body color should be changed to `theme.body` and arrow color should be changed to `theme.text`. Just take a look at the button code I provided in above comment. you will get to understand.

Reply:
Here's my failed attempt:

```
import React from "react";
import "./GoTopButton.css";

const onMouseEnter = (event, color, bgColor) => {
	const el = event.target;
	el.style.color = color;
	el.style.backgroundColor = bgColor;
}

const onMouseOut = (event, color, bgColor) => {
	const el = event.target;
	el.style.color = color;
	el.style.backgroundColor = bgColor;
}

export default function Top({className, theme}) {
    function TopEvent() {
        document.body.scrollTop = 0; // For Safari
        document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
    }
    // When the user scrolls down 20px from the top of the document, show the button   
    function scrollFunction() {
        if (document.body.scrollTop > 50 || document.documentElement.scrollTop > 50) {
            document.getElementById("topButton").style.visibility = "visible";
        } else {
            document.getElementById("topButton").style.visibility = "hidden";
        }
    }
    window.onscroll = function() {scrollFunction();};
    window.onload = function() {scrollFunction();}; //To make sure that this button is not visible at starting.
    // When the user clicks on the button, scroll to the top of the document
    
    return (
        <div className={className}>
			<a
				class="topButton"
				style={{
					color: theme.body,
					backgroundColor: theme.text,
          			border: `solid 1px ${theme.text}`
				}}
				onMouseEnter={(event) => onMouseEnter(event, theme.text, theme.body)}
				onMouseOut={(event) => onMouseOut(event,theme.body, theme.text)}>
			</a>
        <button onClick={TopEvent} id="topButton" title="Scroll up"><i class="fas fa-arrow-up" aria-hidden="true"></i></button>
        </div>
        );
    }
```

Reply:
commit the code here and will see what changes will be needed